### PR TITLE
Fixed PDF download

### PR DESCRIPTION
**Pull request by popkinj**
Wasn't aware the pdf get's generated twice until previous PR broke the download. I've fixed that and tested both back and front end.

### CODE REVIEW

File: fpo-web/src/app/survey/survey.component.ts [hunk 1]
```diff
@@ -271,8 +271,22 @@ export class SurveyComponent implements OnInit, OnDestroy {
   }
 
   submitForm(data: any) {
-    //Default to notice to disputant response. Please change later on. 
-    const url = this.dataService.getApiUrl("submit-form/?name=notice-to-disputant-response");
+    // This is horrendous.
```
Comment (seeker25) [inline fpo-web/src/app/survey/survey.component.ts:274]: Maybe something like this in SurveyJS (Psuedo code, haven't tried):
    surveyModel.onValueChanged.add((sender, options) => {
    });
survey.setVariable("form", value)
survey.onComplete.add(function(survey, options) {
   survey.setValue("form", survey.getVariable("form"));
});

File: fpo-api/api/views/SubmitTicketResponseView.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        return Response({"key": key})
+
+    def post(self, request: Request, name=None):
+        check_captcha = grecaptcha_verify(request)
+        if not check_captcha["status"]:
+            return HttpResponseForbidden(text=check_captcha["message"])
+
+        #############################################################
+        #  Adding different pdf form logic: Jul 3, 2020
+        data = json.loads(request.body)
```
Comment (andrewwhitehead) [inline fpo-api/api/views/SubmitTicketResponseView.py:39]: The request JSON is being loaded twice in this method (also as request.data which is preferred). It should only be loaded once for consistency.

The code to add the date to the payload, select a template, and prepare the PDF should be moved into a separate method and called inside the try block at the end of this method.

File: fpo-api/api/views/SubmitTicketResponseView.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        name = request.query_params.get('name')
+
+        # data = copy.deepcopy(request.data)
```
Comment (andrewwhitehead) [inline fpo-api/api/views/SubmitTicketResponseView.py:42]: Commented lines not needed. The template name should be validated before setting the `template` variable.

File: fpo-api/api/views/SurveyPdfView.py [hunk 4]
```diff
@@ -13,27 +13,87 @@ class SurveyPdfView(generics.GenericAPIView):
     permission_classes = ()  # permissions.IsAuthenticated,)
 
     def post(self, request: Request, name=None):
-        tpl_name = "survey-{}.html".format(name)
-        # return HttpResponseBadRequest('Unknown survey name')
 
-        responses = json.loads(request.POST["data"])
-        # responses = {'question1': 'test value'}
+        #############################################################
```
Comment (andrewwhitehead) [inline fpo-api/api/views/SurveyPdfView.py:17]: Is this view still needed at all?

File: fpo-web/src/app/survey/survey.component.ts [hunk 5]
```diff
@@ -271,8 +271,22 @@ export class SurveyComponent implements OnInit, OnDestroy {
...
+    if (
+      (data.continueDispute == 'y' &&
+      data.disputeType == 'fineAmount' &&
+      data.disputeInWriting == 'y')
+        ||
+      (data.continueDispute == 'n' &&
+      data.moreTimeToPay1 == 'y'
+      )
+    ) {
+      var form = "violation-ticket-statement-and-written-reasons";
```
Comment (andrewwhitehead) [inline fpo-web/src/app/survey/survey.component.ts:284]: Use `let form` before the conditional block and assign `form` inside the branches. `var` is not generally used in typescript code.

**Comment by popkinj** [comment]
OK... I've addressed all raised issues.
I had to make some adjustments to the data transform logic... Because the data being inserted into the templates is the same that's going to the database. So I had to create separate values due to date format issues.

**Comment by seeker25** [comment]
Trying this out now

**Comment by seeker25** [comment]
I'm trying to push some changes, but seem to be lacking some permissions.

**Pull request merged**